yktrace/src/lib.rs

```diff
@@ -65,15 +69,33 @@ impl IRTrace {
         self.blocks.get(idx)
     }
 
-    pub fn compile(&self) {
+    pub fn compile(&self) -> *const c_void {
         let len = self.len();
         let mut func_names = Vec::with_capacity(len);
         let mut bbs = Vec::with_capacity(len);
         for blk in &self.blocks {
             func_names.push(blk.func_name().as_ptr());
             bbs.push(blk.bb());
         }
-        unsafe { ykllvmwrap::__ykllvmwrap_irtrace_compile(func_names.as_ptr(), bbs.as_ptr(), len) }
+
+        // Find bitcode section in executable.
+        let pathb = env::current_exe().unwrap();
+        let file = fs::File::open(&pathb.as_path()).unwrap();
+        let mmap = unsafe { memmap2::Mmap::map(&file).unwrap() };
+        let object = object::File::parse(&*mmap).unwrap();
+        let sec = object.section_by_name(".llvmbc").unwrap();
```

Comment:
Given all these `unwrap`s, I suspect the function signature should be `compile(&self) -> Result<*const c_void, Box<dyn Error>>` and the errors percolated upwards.

Reply:
I'm torn. Part of me thinks that at this early stage unwrapping is ok.

Reply:
Borking on file system accesses is pretty drastic IMHO.

Reply:
This code is gonna go anyway once we load this stuff lazily (and ideally from memory and not the binary).